frontend/pages/sign-in/code.tsx

```diff
@@ -0,0 +1,192 @@
+import { ChangeEvent, ClipboardEvent, useCallback, useEffect } from 'react';
+
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { useRouter } from 'next/router';
+
+import { withLocalizedRequests } from 'hoc/locale';
+
+import { useAppSelector } from 'store/hooks';
+import { session } from 'store/session';
+
+import { InferGetStaticPropsType } from 'next';
+
+import useMe from 'hooks/me';
+
+import { loadI18nMessages } from 'helpers/i18n';
+
+import Alert from 'components/alert';
+import Button from 'components/button';
+import ErrorMessage from 'components/forms/error-message';
+import Input from 'components/forms/input';
+import Loading from 'components/loading';
+import { Paths, UserRoles } from 'enums';
+import NakedLayout, { NakedLayoutProps } from 'layouts/naked';
+import { PageComponent } from 'types';
+import { SignIn, SignInCodeForm } from 'types/sign-in';
+
+import { useSignIn } from 'services/authentication/authService';
+
+export const getStaticProps = withLocalizedRequests(async ({ locale }) => {
+  return {
+    props: {
+      intlMessages: await loadI18nMessages({ locale }),
+    },
+  };
+});
+
+type ProjectDeveloperProps = InferGetStaticPropsType<typeof getStaticProps>;
+
+const CODE_LENGTH = 5;
+
+const SignInCode: PageComponent<ProjectDeveloperProps, NakedLayoutProps> = () => {
+  const { formatMessage } = useIntl();
+  const { replace } = useRouter();
+  const signIn = useSignIn();
+  const { user, isLoading: isUserLoading } = useMe();
+
+  // Email and password got from login page
+  const sessionData = useAppSelector(session);
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setFocus,
+    setValue,
+  } = useForm<SignInCodeForm>({
+    shouldUseNativeValidation: true,
+    reValidateMode: 'onChange',
+  });
+
+  // Redirect if user is signed in
+  useEffect(() => {
```

Comment:
At the moment, anybody can access this page, even if they don't come from the sign in page.